neuraxle/base.py

```diff
@@ -41,8 +41,8 @@ def __init__(
         if hyperparams_space is None:
             hyperparams_space = dict()
 
-        self.hyperparams: HyperparameterSamples = hyperparams
-        self.hyperparams_space: HyperparameterSpace = hyperparams_space
+        self.hyperparams: HyperparameterSamples = HyperparameterSamples(hyperparams)
```

Comment:
Shouldn't we get passed the HyperparameterSamples object instead of creating it here ? I think that constructors should at most assign supplied values to fields, but I could be wrong

Reply:
In the constructor's arguments, we at least say that we expect HyperparameterSamples and HyperparameterSpace types because of the typing annotations. 

This here is just for patching in cases we received a dict. Wouldn't it be fun to be able to just pass in a dict when feeling lazy?